src/Moq/Match.cs

```diff
@@ -33,8 +33,14 @@ internal static TValue Matcher<TValue>()
 
 		internal abstract bool Matches(object value);
 
+		internal virtual void OnSuccess(object value)
+		{
+		}
```

Comment:
You could declare this method `abstract` (like the `Matches` method above it), this empty implementation will never actually get used anywhere as `Match<T>` is (and will most likely remain) the sole subclass of `Match`.

Reply:
I like the idea to leave this method abstract, but it would be a breaking change because `Match` is public.
Are you sure about that?

Reply:
The type may be `public`, but it is impossible to subclass it outside Moq's own assembly due to the presence of existing `internal abstract` members. So I don't see any harm in adding another abstract member.

(Am I missing something else?)